### ISSUE TITLE
Non-working configuration check for more than one module

### ISSUE THREAD

**Issue by vivodikj** [InsertKoinIO/koin-annotations#179]
**Describe the bug**
Koin configuration check don't work for other module as App (presentation) module. Then the Koin works like "OLDER" non annotations version and don't check if we don't forgot something. If we forgot add some of annotations (single, factory...) then we don't get an error but it crash in runtime. The same if we forgot or don't correct create an Module (maybe bad package name and so on).

Error:
e: [ksp] --> Missing Definition type 'com.example.koinannotations.business.features.fake.repository.FakeRepository' for 'com.example.koinannotations.business.features.fake.usecase.CheckIfKoinIsInstalledUseCase'. Fix your configuration to define type 'FakeRepository'.
e: [ksp] --> Missing Definition type 'com.example.koinannotations.business.features.fake.repository.FakeRepository' for 'com.example.koinannotations.business.features.fake.usecase.GetFakeDataUseCase'. Fix your configuration to define type 'FakeRepository'.

**To Reproduce**
My repository: https://gitlab.com/10vivodik/koin-annotations-test
1. Go to 'business module - build.gradle.kts'
2. Scroll down to 'ksp'
3. Uncomment 'arg("KOIN_CONFIG_CHECK","true")' and comment 'arg("KOIN_CONFIG_CHECK","false")'

**Expected behavior**
Koin Annotations processor correctly check every definition in all modules when KOIN_CONFIG_CHECK is set to true in ksp arguments `arg("KOIN_CONFIG_CHECK","true")`. 

**Koin project used and used version:**
 `kotlin = "1.9.24"`
 `ksp = "1.9.24-1.0.20"`
 `koinBom = "3.5.6"`
 `koinAnnotations = "1.4.0-RC4"`

**Additional moduleDefinition**
Module definition is in app/di package `[com.example.koinannotations.app.di]`.

**Comment by arnaudgiuliani** [comment]
I checked your project and found that the dependency configuration is tangled between module compilation and Koin module configuration. Specifically, the way Koin modules are being compiled doesn't align properly with your Gradle module setup.

Issue Overview:
Infrastructure depends on both business and data modules: The problem arises because FakeSource is not defined in these modules—it's defined in the app module.
Configuration Spread: Some Koin definitions are declared globally in the app module, while others are declared locally in the respective modules. This inconsistency is causing cascading configuration issues.

TL;DR:
You need to decide between:

- Restricting compile safety to the main app module, or
- Enabling it across all modules and resolving the configuration issues.

---

Your errors:
```
> Task :data:kspKotlin FAILED
w: [ksp] Check Configuration ...
w: [ksp] Check Configuration ...
e: [ksp] --> Missing Definition type 'com.example.koinannotations.data.source.fake.FakeSource' for 'com.example.koinannotations.data.features.fake.FakeRepositoryImpl'. Fix your configuration to define type 'FakeSource'.
e: Error occurred in KSP, check log for detail

> Task :business-usecase:kspKotlin FAILED
w: [ksp] Check Configuration ...
w: [ksp] Check Configuration ...
e: [ksp] --> Missing Definition type 'com.example.koinannotations.business.features.fake.repository.FakeRepository' for 'com.example.koinannotations.business.features.fake.usecase.CheckIfKoinIsInstalledUseCase'. Fix your configuration to define type 'FakeRepository'.
e: [ksp] --> Missing Definition type 'com.example.koinannotations.business.features.fake.repository.FakeRepository' for 'com.example.koinannotations.business.features.fake.usecase.GetFakeDataUseCase'. Fix your configuration to define type 'FakeRepository'.
e: Error occurred in KSP, check log for detail
```

The FakeRepository definition is missing for use cases relying on it. FakeSource definition is not ready, due to your configuration.

To fix this, you have two main options:

If you want to maintain compile-time safety in all modules: Each module must be self-contained with all required definitions and dependencies. This means extracting shared dependencies (like FakeSource and FakeRepository) into a separate module that all other modules can depend on.

If you prefer simplicity: Keep the compile safety checks isolated in the app module. This approach assumes the app module will act as the central place for dependency definitions and Koin setup. Other modules will not need to validate their definitions individually.

**Comment by arnaudgiuliani** [comment]
this is not a KSP config check issue here 👍
